PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,650 @@
+    public convenience init(apiKey: String,
+                            systemInfo: SystemInfo,
+                            eTagManager: ETagManager,
+                            operationDispatcher: OperationDispatcher) {
+        let httpClient = HTTPClient(systemInfo: systemInfo,
+                                    eTagManager: eTagManager,
+                                    operationDispatcher: operationDispatcher)
+        self.init(httpClient: httpClient, apiKey: apiKey)
+    }
+
+    @objc public required init(httpClient: HTTPClient, apiKey: String) {
+        self.httpClient = httpClient
+        self.apiKey = apiKey
+        self.offeringsCallbacksCache = [:]
+        self.purchaserInfoCallbacksCache = [:]
+    }
+
+    @objc(createAliasForAppUserID:newAppUserID:completion:)
+    public func createAlias(appUserID: String, newAppUserID: String, completion: ((Error?) -> Void)?) {
+        guard let appUserID = try? escapedAppUserID(appUserID: appUserID) else {
+            completion?(ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let path = "/subscribers/\(appUserID)/alias"
+        httpClient.performPOSTRequest(serially: true,
+                                      path: path,
+                                      requestBody: ["new_app_user_id": newAppUserID],
+                                      headers: authHeaders) { statusCode, response, error in
+            self.handle(response: response, statusCode: statusCode, error: error, completion: completion)
+        }
+    }
+
+    @objc public func clearCaches() {
+        httpClient.clearCaches()
+    }
+
+    // swiftlint:disable function_parameter_count
+    @objc (postReceiptData:appUserID:isRestore:productInfo:presentedOfferingIdentifier:observerMode:subscriberAttributes:completion:)
+    public func post(receiptData: Data,
+                     appUserID: String,
+                     isRestore: Bool,
+                     productInfo: ProductInfo?,
+                     presentedOfferingIdentifier offeringIdentifier: String?,
+                     observerMode: Bool,
+                     subscriberAttributes subscriberAttributesByKey: SubscriberAttributeDict?,
+                     completion: @escaping BackendPurchaserInfoResponseHandler) {
+    // swiftlint:enable function_parameter_count
+        let fetchToken = receiptData.base64EncodedString(options: .lineLength64Characters)
+        var body: [String: AnyObject] = [
+            "fetch_token": fetchToken as NSString,
+            "app_user_id": appUserID as NSString,
+            "is_restore": NSNumber(value: isRestore),
+            "observer_mode": NSNumber(value: observerMode),
+        ]
+
+        let cacheKey =
+        """
+        \(appUserID)-\(isRestore)-\(fetchToken)-\(productInfo?.cacheKey ?? "")
+        -\(offeringIdentifier ?? "")-\(observerMode)-\(subscriberAttributesByKey?.debugDescription ?? "")"
+        """
+
+        if add(callback: completion, key: cacheKey) {
```

Comment:
I realize that this is an exact port of what we had before, but I had to go in and check the implementation to understand what it does. 

Perhaps we could rename this method? I'm honestly struggling to come up with a good name too, since this adds the value to the completion blocks for the key if needed and returns a bool stating whether a request was already in flight. 

`addCallbackToListAndReturnIfRequestNotInFlight` doesn't quite roll off the tongue or describe what's happening. 

what if we had a private enum, i.e.: 
```swift
let result = appendToCallbacksList(callback, forKey: cacheKey)
if result == .requestAlreadyInFlight { 
    return
}
```

Reply:
I'll split the difference.